src/react/hooks/__tests__/useBackgroundQuery.test.tsx

```diff
@@ -0,0 +1,2290 @@
+      ];
+
+      cache.writeQuery({
+        query,
+        data: { results },
+      });
+
+      const suspenseCache = new SuspenseCache();
+
+      const { result } = renderHook(
+        () =>
+          useBackgroundQuery(query, {
+            canonizeResults: true,
+          }),
+        {
+          wrapper: ({ children }) => (
+            <MockedProvider cache={cache} suspenseCache={suspenseCache}>
+              {children}
+            </MockedProvider>
+          ),
+        }
+      );
+
+      const { queryRef } = result.current;
+
+      const _result = await queryRef.promises.main;
+      const resultSet = new Set(_result.data.results);
+      const values = Array.from(resultSet).map((item) => item.value);
+
+      expect(_result.data).toEqual({ results });
+      expect(_result.data.results.length).toBe(6);
+      expect(resultSet.size).toBe(5);
+      expect(values).toEqual([0, 1, 2, 3, 5]);
+    });
+
+    it("can disable canonical results when the cache's canonizeResults setting is true", async () => {
+      interface Result {
+        __typename: string;
+        value: number;
+      }
+
+      const cache = new InMemoryCache({
+        canonizeResults: true,
+        typePolicies: {
+          Result: {
+            keyFields: false,
+          },
+        },
+      });
+
+      const query: TypedDocumentNode<{ results: Result[] }> = gql`
+        query {
+          results {
+            value
+          }
+        }
+      `;
+
+      const results: Result[] = [
+        { __typename: 'Result', value: 0 },
+        { __typename: 'Result', value: 1 },
+        { __typename: 'Result', value: 1 },
+        { __typename: 'Result', value: 2 },
+        { __typename: 'Result', value: 3 },
+        { __typename: 'Result', value: 5 },
+      ];
+
+      cache.writeQuery({
+        query,
+        data: { results },
+      });
+
+      const suspenseCache = new SuspenseCache();
+
+      const { result } = renderHook(
+        () =>
+          useBackgroundQuery(query, {
+            canonizeResults: false,
+          }),
+        {
+          wrapper: ({ children }) => (
+            <MockedProvider cache={cache} suspenseCache={suspenseCache}>
+              {children}
+            </MockedProvider>
+          ),
+        }
+      );
+
+      const { queryRef } = result.current;
+
+      const _result = await queryRef.promises.main;
+      const resultSet = new Set(_result.data.results);
+      const values = Array.from(resultSet).map((item) => item.value);
+
+      expect(_result.data).toEqual({ results });
+      expect(_result.data.results.length).toBe(6);
+      expect(resultSet.size).toBe(6);
+      expect(values).toEqual([0, 1, 1, 2, 3, 5]);
+    });
+  });
+
+  describe('fetch policy behaviors', () => {
+    describe.skip('cache-and-network', () => {
+      // TODO: should return cache data first if it exists
+      it('returns initial cache data followed by network data', async () => {
```

Comment:
As mentioned before, I generally prefer tests to be as [flat](https://kentcdodds.com/blog/avoid-nesting-when-youre-testing) as possible. 

One of the biggest reasons for this is that as I'm reading a test and understanding what we are looking at, I use the test description as a means to understand what the test is meant for. In a lot of tests where I start to see nested `describes`, the description starts to get split between the `describe` and the `it`/`test` description. This gets problematic the more the file grows since it now requires me to scroll to understand the whole context for the test.

Instead, I prefer more descriptive tests with a flat structure. This allows me to stay isolated to the test without the need to scroll around to find additional context.

For this test, that would include adding the `cache-and-network` qualifier in the test description itself.

```suggestion
      it('returns initial cache data followed by network data when the fetch policy is `cache-and-network`', async () => {
```

I know its longer and some of it is repetitive, but again, I can look at this test and not have to scroll around to find additional context.

Thoughts?

Reply:
Updated test name in https://github.com/apollographql/apollo-client/pull/10755/commits/5b1d3ee60a81ad9b361ef79eb857d3079c234144